x/nexus/keeper/grpc_query.go

```diff
@@ -88,19 +88,24 @@ func (k Keeper) TransferFee(c context.Context, req *types.TransferFeeRequest) (*
 		return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered chain", req.DestinationChain)
 	}
 
-	if !k.IsAssetRegistered(ctx, sourceChain, req.Amount.Denom) {
-		return nil, sdkerrors.Wrapf(types.ErrNexus, "%s is not a registered asset on chain %s", req.Amount.Denom, sourceChain.Name)
+	amount, err := req.GetAmount()
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "invalid amount")
```

Comment:
you already wrap inside the GetAmount with this description

Reply:
Fixed